app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java

```diff
@@ -22,8 +26,21 @@
 import org.mozilla.focus.search.SearchEngineManager;
 import org.mozilla.focus.telemetry.TelemetryWrapper;
 import org.mozilla.focus.utils.SupportUtils;
+import org.mozilla.focus.utils.UrlUtils;
+import org.mozilla.focus.utils.ViewUtils;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
 
 public class ManualAddSearchEngineSettingsFragment extends SettingsFragment {
+    private static String LOGTAG = "ManualAddSearchEngine";
+
+    // Set so the user doesn't have to wait *too* long. It's used twice: once for connecting and once for reading.
+    private static int SEARCH_QUERY_VALIDATION_TIMEOUT_MILLIS = 4000;
```

Comment:
Is 4 seconds too long?

Reply:
I didn't want to prevent users with poor internet connections from using this feature and tried to balance it with how long users will realistically wait.